### ISSUE TITLE
bug: using pydocstyle (D) does not enable D417

### ISSUE THREAD

**Issue by bp50hz** [astral-sh/ruff#15065]
### Description of the bug
selecting pydocstyle (D) does not enable D417
issue might be linked to using 'convention = "numpy"'

### To Reproduce

ruff.toml:
```

[lint]
select = [ "D"]

[lint.pydocstyle]
convention = "numpy"

```

example.py:
(docstring error see var2 -> var3)
```

'''Example module that contains a function with type hints.'''


def example(var1: int, var2: str) -> str:
    '''Concat variables.

    Parameters
    ----------
    var1 : int
        This is an integer argument
    var3 : str
        This is a string argument

    Returns
    -------
    str
        This is a string that contains the values of the two arguments
    '''
    return f"var1: {var1}, var2: {var2}"

```
```
ruff check
All checks passed!
```
(Expected a D417 error)

change ruff.toml:
```
[lint]
select = [ "D", "D417"]

[lint.pydocstyle]
convention = "numpy"

```
or
change ruff.toml:
```
[lint]
select = [ "D"]
```

```
ruff check
D417 Missing argument description in the docstring for `example`: `var2`
  |
4 | def example(var1: int, var2: str) -> str:
  |     ^^^^^^^ D417
5 |     """Concat variables.
  |
```
(Expected D417 Error showed up)

Also: should there not be an error for "var3" in the docstring?

### Expected behavior
Selecting a rule enables all stable rules of the set.

### Environment information
ruff 0.8.3
Python 3.10.15

**Comment by dylwil3** [comment]
Yep - this rule is currently only supported for the google convention. Apologies!

From [the docs](https://docs.astral.sh/ruff/rules/undocumented-param/):

> This rule is enabled when using the google convention, and disabled when using the pep257 and numpy conventions.

**Comment by bp50hz** [comment]
It would be nice to get some level of information about this. Are there any other excluded? Are there any other Rules with this behavior? When I'm using one Rule I don't always want to second guess if it uses all the subrules.

Thank you for your time

**Comment by MichaReiser** [comment]
That's understandable. The [FAQ section](https://docs.astral.sh/ruff/faq/#does-ruff-support-jupyter-notebooks) lists the enabled rules by convention
